packages/protocol/contracts/L1/libs/LibVerifying.sol

```diff
@@ -113,59 +116,60 @@ library LibVerifying {
         TaikoData.ProposedBlock storage proposal
     ) private returns (ChainData memory chainData) {
         if (config.enableTokenomics) {
-            uint256 newFeeBase;
-            {
-                // otherwise: stack too deep
-                uint256 reward;
-                uint256 tRelBp; // [0-10000], see the whitepaper
-                (newFeeBase, reward, tRelBp) = LibTokenomics.getProofReward({
+            (uint256 newFeeBase, uint256 reward, uint256 tRelBp) = LibTokenomics
+                .getProofReward({
                     state: state,
                     config: config,
                     provenAt: fc.provenAt,
                     proposedAt: proposal.proposedAt
                 });
 
-                // reward the prover
-                if (reward > 0) {
-                    if (state.balances[fc.prover] == 0) {
-                        // Reduce reward to 1 wei as a penalty if the prover
-                        // has 0 TKO outstanding balance.
-                        state.balances[fc.prover] = 1;
-                    } else {
-                        state.balances[fc.prover] += reward;
-                    }
+            // reward the prover
+            if (reward > 0) {
+                if (state.balances[fc.prover] == 0) {
+                    // Reduce reward to 1 wei as a penalty if the prover
+                    // has 0 TKO outstanding balance.
+                    state.balances[fc.prover] = 1;
```

Comment:
Whoever is running the block verification still pays the same amount of gas, so maybe better to simply not do anything? This will reduce the worst case gas cost for block verification.